<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsSQLComposerDialogBase</class>
 <widget class="QDialog" name="QgsSQLComposerDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>747</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SQL query composer</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_7">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="mSQLStatementLabel">
         <property name="text">
          <string>SQL statement</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QgsCodeEditorSQL" name="mQueryEdit" native="true">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This is the SQL query editor.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="0,0">
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item>
      <layout class="QFormLayout" name="formLayout">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="mColumnsLabel">
         <property name="text">
          <string>Columns</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QTextEdit" name="mColumnsEditor">
         <property name="minimumSize">
          <size>
           <width>400</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="mTablesLabel">
         <property name="text">
          <string>Table(s)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="mTablesEditor">
         <property name="text">
          <string/>
         </property>
         <property name="frame">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="mJoinsLabels">
         <property name="text">
          <string>Joins</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <widget class="QTableWidget" name="mTableJoins">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>398</width>
             <height>0</height>
            </size>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="rowCount">
            <number>0</number>
           </property>
           <attribute name="horizontalHeaderDefaultSectionSize">
            <number>200</number>
           </attribute>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>false</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Joint layer</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>ON condition</string>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="mAddRemoveLayout">
           <item>
            <widget class="QPushButton" name="mAddJoinButton">
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../../images/images.qrc">
               <normaloff>:/images/themes/default/symbologyAdd.png</normaloff>:/images/themes/default/symbologyAdd.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="mRemoveJoinButton">
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../../images/images.qrc">
               <normaloff>:/images/themes/default/symbologyRemove.png</normaloff>:/images/themes/default/symbologyRemove.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="mAddRemoveSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>1</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="mWhereLabel">
         <property name="text">
          <string>Where    </string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QTextEdit" name="mWhereEditor"/>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="mOrderLabel">
         <property name="text">
          <string>Order by</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QTextEdit" name="mOrderEditor">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="verticalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QToolBox" name="mToolbox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>250</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>250</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="mPageData">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>250</width>
            <height>495</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Data</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QComboBox" name="mTablesCombo">
             <item>
              <property name="text">
               <string>Tables</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="mColumnsCombo">
             <item>
              <property name="text">
               <string>Columns</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="Line" name="line_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="mAggregatesCombo">
             <item>
              <property name="text">
               <string>Aggregates</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="mFunctionsCombo">
             <item>
              <property name="text">
               <string>Functions</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="mSpatialPredicatesCombo">
             <item>
              <property name="text">
               <string>Spatial predicates</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="mStringFunctionsCombo">
             <item>
              <property name="text">
               <string>Strings functions</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="mOperatorsCombo">
             <item>
              <property name="text">
               <string>Operators</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="mPageColumnsValues">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>250</width>
            <height>495</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Columns' values</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QComboBox" name="columns_2">
             <item>
              <property name="text">
               <string>Columns</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="extract">
             <property name="text">
              <string>Only 10 first values</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QListView" name="values">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <property name="showDropIndicator" stdset="0">
              <bool>false</bool>
             </property>
             <property name="dragEnabled">
              <bool>false</bool>
             </property>
             <property name="dragDropMode">
              <enum>QAbstractItemView::NoDragDrop</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QDialogButtonBox" name="mButtonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Help|QDialogButtonBox::Ok|QDialogButtonBox::Reset</set>
       </property>
       <property name="centerButtons">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCodeEditorSQL</class>
   <extends>QWidget</extends>
   <header>qgscodeeditorsql.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../images/images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>mButtonBox</sender>
   <signal>rejected()</signal>
   <receiver>QgsSQLComposerDialogBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>633</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mButtonBox</sender>
   <signal>accepted()</signal>
   <receiver>QgsSQLComposerDialogBase</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>516</x>
     <y>433</y>
    </hint>
    <hint type="destinationlabel">
     <x>306</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
